fix(action-popover): submenu stays open when another submenu is clicked

mihai-albu-sage opened the pull request:
### Proposed behaviour

We are now maintaining the selected submenu in the action popover context. On state update, we compare the current value and automatically close any submenus that don't match, ensuring only one submenu remains open at a time regardless if it was opened by click event or keyboard event.

https://github.com/user-attachments/assets/8bd5a306-567f-43a9-86c1-26700281147a

<!--
A clear and concise description of what changes this PR makes. If applicable, include any UI screenshots to help explain your request. If you are a Sage contributor, please DO NOT share any commercially sensitive information, such as UI screenshots or code from Sage products.
-->

### Current behaviour

Currently if a submenu is opened by keyboard event, and you open a new submenu by mouse event, the first submenu is not closed.

https://github.com/user-attachments/assets/90ae0580-57f4-49c6-aa60-aabec2283f20

<!--
A clear and concise description of the behaviour before this change. If applicable, include any UI screenshots to help explain your request. If you are a Sage contributor, please DO NOT share any commercially sensitive information, such as UI screenshots or code from Sage products.
-->

### Checklist

<!-- Each PR should include the following -->

- [X] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [X] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [X] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

<!-- Add any other context or links about the pull request here. -->

### Testing instructions

1. Go to the default action popover storybook
2. Open the first action popover using the keyboard
3. Use the keyboard to navigate to the first submenu and open it
4. Hover the second submenu with the mouse
5. Click the second submenu to open it

Actual : the first submenu does not close
Expected: the first submenu should close after clicking the second submenu

<!--
How can a reviewer test this PR?

If this PR addresses a pre-existing bug, please include a link to a sandbox that reproduces the original bug. A starter template has been provided to help you do this:
<https://stackblitz.com/fork/github/Parsium/carbon-starter>
-->

nuria1110 reviewed (commented):
Changes look good to me, however we may also want to also close the submenu if another is opened on hover as I don't believe they should be able to overlap, similar to how we do in `Menu`.

mihai-albu-sage commented:
> Changes look good to me, however we may also want to also close the submenu if another is opened on hover as I don't believe they should be able to overlap, similar to how we do in `Menu`.

Refactored the code to close the submenu when another submenu is opened by hovering. Also added a playwright test to capture this behaviour.
